data/data_interface.py

```diff
@@ -0,0 +1,160 @@
+                target_feature : str, target_mapping : Any = None, scaling_method : str = "MinMax", encoding_method : str = "OneHot"):
+        """
+        Creates a data interface with the specified, continuous features, ordinal features, and categorical features.
+
+        Features can be normalized, standardized, or encoded as desired. Additionally data can be split into train and test data.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            A dataframe of the data to load. If None, we will attempt to load using file_path
+        file_path : str
+            The path to load your data from. Not required with data is not None.
+        continuous_features : Sequence[str]
+            A sequence of the continous features.
+        ordinal_features : Sequence[str]
+            A sequence of the ordinal features.
+        categorical_features : Sequence[str]
+            A sequence of the categorical/nominal features.
+        immutable_features : Sequence[str]
+            A sequence of the immutable features.
+        target_feature : str
+            The column name of the target feature
+        target_mapping : Any
+            A mapping of the target features to some values
+        scaling_method : str
+            String indicating how you with to scale your data. Options are MaxMin normalization and standardization 
+        encoding_method : str
+            Encoding method for categorical features. Only one-hot for now.
+        """
+        self._continuous_features = continuous_features
+        self._ordinal_features =  ordinal_features
+        self._categorical_features = categorical_features
+        self._immutable_features = immutable_features
+        self._target_feature = target_feature
+        self._target_mapping = target_mapping
+        self._scaling_method = scaling_method
+        self._encoding_method = encoding_method
+        if (file_path is None or file_path is "") and isinstance(data, pd.DataFrame):
+            df = data
+        elif file_path is not None and file_path is not "":
+            df = pd.read_csv(file_path)
+        else:
+            raise Exception("No data and file path provided")
+        X_features = df.columns[df.columns != target_feature]
+        self._df_y = df[self._target_feature]
+        self._df_X = df[X_features]
+    
+    def split_data(self, test_size : float = 0.3) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame ,pd.DataFrame]:
+        """
+        Splits data into X and y and train and test data. Should be used as the last step when setting up data.
+
+        Parameters
+        ----------
+        test_size : float
+            Between 0.0 and 1.0 and represents the proportion of the dataset to include in the test split.
+        Returns
+        -------
+        X_train, X_test, y_train, y_test : Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame ,pd.DataFrame]
```

Comment:
Prefer 'data' and 'labels' in place of X and y. (https://android.googlesource.com/platform/external/google-styleguide/+/refs/tags/android-s-beta-2/pyguide.md 3.16 Naming)